components/fxa-client/src/auth.rs

```diff
@@ -12,32 +12,23 @@
 //!
 //!    - A traditional OAuth flow, where the user is directed to a webpage to enter
 //!      their account credentials and then redirected back to the application.
-//!      This is exposed by the [`begin_oauth_flow`](FirefoxAccount::begin_oauth_flow)
+//!      This is exposed by the [`connect_with_oauth`](FirefoxAccount::connect_with_oauth)
 //!      method.
 //!
 //!    - A device pairing flow, where the user scans a QRCode presented by another
 //!      app that is already connected to the account, which then directs them to
 //!      a webpage for a simplified signing flow. This is exposed by the
-//!      [`begin_pairing_flow`](FirefoxAccount::begin_pairing_flow) method.
+//!      [`connect_with_pairing`](FirefoxAccount::connect_with_pairing) method.
 //!
 //! Technical details of the pairing flow can be found in the [Firefox Accounts
 //! documentation hub](https://mozilla.github.io/ecosystem-platform/docs/features/firefox-accounts/pairing).
 
-use crate::{ApiResult, Error, FirefoxAccount};
+use crate::{ApiResult, CallbackResult, Error, FirefoxAccount};
 use error_support::handle_error;
 use std::collections::HashMap;
 
 impl FirefoxAccount {
-    /// Initiate a web-based OAuth sign-in flow.
-    ///
-    /// This method initializes some internal state and then returns a URL at which the
-    /// user may perform a web-based authorization flow to connect the application to
-    /// their account. The application should direct the user to the provided URL.
-    ///
-    /// When the resulting OAuth flow redirects back to the configured `redirect_uri`,
-    /// the query parameters should be extracting from the URL and passed to the
-    /// [`complete_oauth_flow`](FirefoxAccount::complete_oauth_flow) method to finalize
-    /// the signin.
```

Comment:
We're losing some context by deleting those comments, we should document at least the expectation that we are enforcing in our oauth handler. 

I'm not sure if documenting them here as doc comments is best (and I know we have some notes on the callback interface itself) - but we should make sure the context isn't fully lost

Reply:
I think the documentation in `internal/auth.rs` is decent.  It describes both halves of the operation in `begin_oauth_flow()/begin_pairing_flow()` and `complete_oauth_flow()` .  Is there something I should add there?

Reply:
Not sure if it belongs here, but possibly belongs at whatever layer the consumer will be blocked. The message we should send is "This API will block until the handler given in the constructor is completed", I understand that we have that information in the `OAuthHandler` itself - but it might be beneficial to have it on the API itself as a warning